### PR TITLE
Migrate importer folder to the project folder structure

### PR DESCRIPTION
**Pull request by domupls**

### CODE REVIEW

File: python/scripts/process_osm.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    if len(sys.argv) < 2 or (tag:=sys.argv[1]) in ["-h", "--help"]:
+        display_help()
+
+    elif len(sys.argv) < 3:
+        raise MissingInputError(f"Insufficient arguments. Use \"{os.path.basename(__file__)} -h/--help\" for hint.")
+    elif not os.path.exists((input_file:=sys.argv[2])):
+        raise FileNotFoundError(f"File '{input_file}' does not exist.")
+    elif not is_valid_extension(input_file):
+        raise InvalidInputError(f"File must have one of the following extensions: osm, osm.pbf, osm.bz2")
+    elif tag == "-d":
```
Comment (marekcuchy) [inline python/scripts/process_osm.py:85]: Please add a brief description of what each case is handling. While the error handles are quite clear from the messages, the rest is not so much.

File: python/tests/data/test.osm
Comment (marekcuchy) [inline python/tests/data/test.osm:1]: The other test files are well-named. This file is used to test what?

**Review by marekcuchy — CHANGES_REQUESTED** [review]

File: python/scripts/process_osm.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    print("   -b        : Extract greatest bounding box from given relation ID of input_file and upload to PostgreSQL database using osm2pgsql.")
+    print("               (Optional: specify style file - default.lua is used otherwise)")
+
+def extract_bbox(relation_id):
+    """Function to determine bounding box"""
+    content = load_multipolygon_by_id(relation_id)
+    min_lon, min_lat, max_lon, max_lat = find_min_max(content)
+    return min_lon, min_lat, max_lon, max_lat
+
+def process_osm_command(tag, input_file, output_file):
```
[the remark this answers is not in the document]
Comment (marekcuchy) [reply python/scripts/process_osm.py:41]: or should it be: run_osm_processing?

File: python/scripts/filter_osm.py [hunk 3]
```diff
@@ -12,111 +12,132 @@ class MissingInputError(Exception):
...
+
+    parser.format_help = lambda: display_help()
+    args = parser.parse_args(arg_list)
+
+    return args
+
+def main(arg_list: list[str] | None = None):
+    args = parse_args(arg_list)
+
+    if not args.input_file:
```
Comment (marekcuchy) [inline python/scripts/filter_osm.py:108]: This check (input_file is set) should be possible to do with the library. Usually, the parsing libraries allow the marking of specific args as mandatory and throw an exception if not set (probably it already does that already).

File: python/scripts/filter_osm.py [hunk 4]
```diff
@@ -12,111 +12,132 @@ class MissingInputError(Exception):
...
+    -R                   : Omit referenced objects (optional for 'f' tag)"""
+    print(help_text)
+
+def is_valid_extension(file: str):
+    """Function to check if the file has a valid extension."""
     valid_extensions = ["osm", "osm.pbf", "osm.bz2"]
     return any(file.endswith(f".{ext}") for ext in valid_extensions)
 
-def check_strategy(strategy):
+def check_strategy(strategy: str | None):
```
Comment (marekcuchy) [inline python/scripts/filter_osm.py:36]: This function never returns false. It should return nothing instead. The usage of it within the if statement misleads that there is the possibility that it returns false.

File: python/scripts/filter_osm.py [hunk 5]
```diff
@@ -12,111 +12,132 @@ class MissingInputError(Exception):
...
+    b                    : Filter geographic objects based on bounding box (with osmium)
+    f                    : Filter objects based on tags in expression_file
+ Option:
+    -c [bbox]            : Specify bouding box (required for 'b' tag)
+                        (Bounding box is specified directly or in config geojson file)
+    -e [expression_file] : Specify path to expression file (required for 't' tag)
+    -r [relation_id]     : Specify relation_id (required for 'id' tag)
+    -s [strategy]        : Specify strategy type (optional for: 'id', 'b')
+    -R                   : Omit referenced objects (optional for 'f' tag)"""
+    print(help_text)
```
Comment (marekcuchy) [inline python/scripts/filter_osm.py:29]: You can generate this message "automatically" with the library.

File: python/scripts/filter_osm.py [hunk 6]
```diff
@@ -12,111 +12,132 @@ class MissingInputError(Exception):
...
+        if not args.expression_file:
+            raise MissingInputError("Expression file needs to be specified.")
+        elif not os.path.exists(args.expression_file):
+            raise FileNotFoundError(f"File '{args.expression_file}' does not exist.")
+
+        run_osmium_filter(args.input_file, args.expression_file, args.omit_referenced)
+
+if __name__ == '__main__':
+    # main()
+    args = parse_args(["b", "input.osm", "10,20,30,40"])
```
Comment (marekcuchy) [inline python/scripts/filter_osm.py:142]: This is probably just testing setup?

File: python/scripts/filter_osm.py [hunk 7]
```diff
@@ -12,111 +12,132 @@ class MissingInputError(Exception):
...
+    args = parse_args(arg_list)
+
+    if not args.input_file:
+        raise MissingInputError("Input file not provided.")
+    elif not os.path.exists(args.input_file):
+        raise FileNotFoundError(f"File '{args.input_file}' does not exist.")
+    elif not is_valid_extension(args.input_file):
+        raise InvalidInputError("File must have one of the following extensions: osm, osm.pbf, osm.bz2")
+    
+    if args.tag == "id":
```
Comment (marekcuchy) [inline python/scripts/filter_osm.py:115]: change this chain of elif to a simple switch.

File: python/scripts/process_osm.py [hunk 8]
```diff
@@ -1,116 +1,133 @@
...
+    r                : Renumber object IDs in OSM file
+    sr               : Sort and renumber objects in OSM file
+    u                : Upload OSM file to PostgreSQL database using osm2pgsql.
+    b                : Extract greatest bounding box from given relation ID of input_file 
+                       and upload to PostgreSQL database using osm2pgsql
+Option:
+    -r [relation_id] : Specify relation_id (required for 'b' tag)
+    -l [style_file]  : Specify style_file (optional for: 'u', 'b' tag) - default.lua is used otherwise
+    -o [output_file] : Specify output_file (required for 's', 'r', 'sr' tags)"""
+    print(help_text)
```
Comment (marekcuchy) [inline python/scripts/process_osm.py:27]: Use the library to print the help message

File: python/scripts/process_osm.py [hunk 9]
```diff
@@ -1,116 +1,133 @@
...
+        raise FileNotFoundError(f"File '{args.input_file}' does not exist.")
+    elif not is_valid_extension(args.input_file):
+        raise InvalidInputError("File must have one of the following extensions: osm, osm.pbf, osm.bz2.")
+    elif args.style_file:
+        if not os.path.exists(args.style_file):
+            raise FileNotFoundError(f"File '{args.style_file}' does not exist.")
+        elif not args.style_file.endswith(".lua"):
+            raise InvalidInputError("File must have the '.lua' extension.")
+    
+    if args.tag in ['d', 'i', 'ie']:
```
Comment (marekcuchy) [inline python/scripts/process_osm.py:109]: use switch. Multiple cases can be done with: `case 'd' | 'i' | 'ie'`

**Review by marekcuchy — CHANGES_REQUESTED** [review]

File: python/scripts/filter_osm.py [hunk 10]
```diff
@@ -89,55 +71,58 @@ def run_osmium_filter(input_file: str, expression_file: str, omit_referenced: bo
...
+                        help="""
+id : Filter geographic objects based on relation ID
+b  : Filter geographic objects based on bounding box (with osmium)
+f  : Filter objects based on tags in expression_file
+""")
     parser.add_argument('input_file', nargs='?', help='Path to input OSM file')
     parser.add_argument("-e", dest="expression_file", nargs="?", help="Path to expression file for filtering tags (required for 'f' tag)")
     parser.add_argument("-c", dest="coords", nargs="?",help="Bounding box coordinates or path to config file (required for 'b' tag)")
-    parser.add_argument("-r", dest="relation_id", nargs="?", help="relation ID (required for 'b' tag)")
+    parser.add_argument("-id", dest="relation_id", nargs="?", help="relation ID (required for 'b' tag)")
```
Comment (marekcuchy) [inline python/scripts/filter_osm.py:83]: The tag 'id' and the argument '-id' are slightly confusing. Consider change.

File: resources/lua_styles/default.lua [hunk 11]
```diff
@@ -50,9 +50,10 @@ end
 
 -- Process every way in the input
 function osm2pgsql.process_way(object)
-    if clean_tags(object.tags) then
-        return
-    end
+    -- if clean_tags(object.tags) then
+    --     return
+    -- end
+    clean_tags(object.tags)
```
Comment (marekcuchy) [inline resources/lua_styles/default.lua:56]: Delete the commented code. You can use git to find it later. The same below.

**Review by marekcuchy — CHANGES_REQUESTED** [review]

**Comment by domupls** [comment]
I've also implemented args parser (#65bac23) to `main.py` to replace using of sys.argv

**Review by marekcuchy — APPROVED** [review]

**Review by marekcuchy — APPROVED** [review]

**Review by marekcuchy — APPROVED** [review]

**Pull request merged**